### PR TITLE
test: add test ranking evaluator with score field

### PR DESCRIPTION
**Pull request by JoanFM**
**Changes introduced**
Have test that will give more confidence that https://github.com/jina-ai/jina-hub/pull/461 will work

### CODE REVIEW

File: tests/unit/drivers/test_rankingevaluation_driver.py [hunk 1]
```diff
@@ -170,3 +171,66 @@ def test_evaluate_assert_doc_groundtruth_structure(simple_chunk_evaluate_driver,
...
+        actual_at_k = actual[:self.eval_at]
+        desired_at_k = desired[:self.eval_at]
+        if len(actual) < 2:
+            raise ValueError(f'Expecting gains with minimal length of 2, {len(actual)} received.')
+        dcg = _compute_dcg(gains=actual_at_k)
+        idcg = _compute_idcg(gains=desired_at_k)
+        if idcg == 0.0:
+            return 0.0
+        else:
+            return dcg / idcg
```
Comment (bhavsarpratik) [inline tests/unit/drivers/test_rankingevaluation_driver.py:214]:
```suggestion
        return 0.0 if idcg == 0.0 else dcg / idcg
```

File: tests/unit/drivers/test_rankingevaluation_driver.py [hunk 2]
```diff
@@ -170,3 +171,66 @@ def test_evaluate_assert_doc_groundtruth_structure(simple_chunk_evaluate_driver,
...
+    ret = 0.0
+    for score, position in zip(gains[1:], range(2, len(gains) + 1)):
+        ret += score / log(position, 2)
+    return gains[0] + ret
+
+
+def _compute_idcg(gains):
+    """Compute ideal discounted cumulative gain."""
+    sorted_gains = sorted(gains, reverse=True)
+    return _compute_dcg(sorted_gains)
```
Comment (bhavsarpratik) [inline tests/unit/drivers/test_rankingevaluation_driver.py:188]: Why do we have NDCG logic in the test?

Comment (JoanFM) [reply tests/unit/drivers/test_rankingevaluation_driver.py:188]: just for a test, I would like to have something more stupid there, can u think of some other stupid measure we can take based on scores? I also do not like having this logic here.

Comment (bhavsarpratik) [reply tests/unit/drivers/test_rankingevaluation_driver.py:188]: I think precision_at_k is the dumbest and can be taken from [here](https://gist.github.com/bwhite/3726239)

Comment (JoanFM) [reply tests/unit/drivers/test_rankingevaluation_driver.py:188]: The point is that precision is based on ids of matches while ndcg works on relevancy scores. That is the difference I want to test, but anyway is an unnecesssry test. Will close

**Pull request closed without merging**